File: node/models/delivery.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Delivery = sequelize.define('Delivery', {
    id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
    delivery_order_number: { type: DataTypes.STRING(100), allowNull: false, unique: true },
    part_number: { type: DataTypes.STRING(100), allowNull: false },
    quantity_shipped: { type: DataTypes.DECIMAL(10, 2), allowNull: false },
    delivery_date: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
    user_id: { type: DataTypes.INTEGER },
    customer: { type: DataTypes.STRING(255) },
    notes: { type: DataTypes.TEXT }
}, { tableName: 'delivery', timestamps: true });

module.exports = Delivery;

File: app/models/delivery.py
from sqlalchemy import Column, Integer, String, Text, TIMESTAMP, DECIMAL, ForeignKey
from sqlalchemy.orm import relationship
from app.database.base import Base

class Delivery(Base):
    __tablename__ = "delivery"
    id = Column(Integer, primary_key=True, index=True)
    delivery_order_number = Column(String(100), unique=True, nullable=False)
    part_number = Column(String(100), ForeignKey("master_prod.part_number"))
    quantity_shipped = Column(DECIMAL(10, 2), nullable=False)
    delivery_date = Column(TIMESTAMP)
    user_id = Column(Integer, ForeignKey("users.id"))
    customer = Column(String(255))
    notes = Column(Text)

    user = relationship("User")
    product = relationship("MasterProd")
